### PR TITLE
*: allow to build NeoGo binaries on Windows

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Related to #2008.

This PR fixes compilation errors and allows to build NeoGo node with `GOOS=windows` and `GOARCH=amd64` envs. The resulting binary is runnable (tested `node` command on the current testnet/mainnet), but the most of the tests are failing, the issue needs additional investigation.

At least, we're able to build/run the node, and it works somehow.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 BRANCH = "master"
 REPONAME = "neo-go"
 NETMODE ?= "privnet"
-BINARY = "./bin/neo-go"
+BINARY = '$(shell if [ "$(shell go env GOOS)" = "windows" ]; then echo "./bin/neo-go.exe"; else echo "./bin/neo-go"; fi)'
```
Comment (roman-khimov) [inline Makefile:4]: How about this?
```
EXTENSION=
ifeq ($(GOOS),windows)
  EXTENSION=".exe"
endif
BINARY=./bin/neo-go$(EXTENSION)
```

File: README.md [hunk 2]
```diff
@@ -56,6 +56,19 @@ make build
...
+To build NeoGo on Windows platform we recommend you to install `make` from [MinGW
+package](https://osdn.net/projects/mingw/). Then you can build NeoGo with:
+
+```
+make build
+```
+
+The resulting binary is `bin/neo-go.exe`.
+
+We also recommend you to switch the Windows Brandmauer off for further NeoGo node run.
```
Comment (roman-khimov) [inline README.md:70]: Firewall.

File: Makefile (hunk 1 above)
Comment (AnnaShaleva) [reply Makefile]: `$(GOOS)` doesn't work for me neither from Makefile, nor from the windows shell, but `$(shell go env GOOS)` works fine. Maybe I have the wrong ENV settings.

Fixed.

Comment (roman-khimov) [reply Makefile]: Yeah, it's probably not a part of regular environment, so you get it from `go`.

File: Makefile [hunk 3]
```diff
@@ -1,7 +1,11 @@
 BRANCH = "master"
 REPONAME = "neo-go"
 NETMODE ?= "privnet"
-BINARY = "./bin/neo-go"
+EXTENSION=
+ifeq ($(shell go env GOOS),windows)
```
Comment (fyrchik) [inline Makefile:5]: This won't work when GOOS is provided via CLI arguments: `make GOOS=windows`. Though I am not sure if anyone use it like this at all and `GOOS=windows make` works anyway.

Comment (AnnaShaleva) [reply Makefile:5]: Well, the most fearest thing with it is that the resulting binary will have no extension. I think if user builds it with `make GOOS=windows` then he can add the proper extension by itself.

Comment (fyrchik) [reply Makefile:5]: We can do something like `GOOS ?= $(shell go env GOOS)` and check for `$(GOOS)` which should work for all cases.

Comment (AnnaShaleva) [reply Makefile]: Nice solution, implemented.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**